lib/puppet/pops/types/p_object_type.rb

```diff
@@ -437,7 +437,7 @@ def initialize(_pcore_init_hash, init_hash_expression = nil)
   end
 
   def instance?(o, guard = nil)
-    if o.is_a?(PuppetObject)
+    if o.instance_of?(PuppetObject)
       assignable?(o._pcore_type, guard)
```

Comment:
I understand calling `instance?` when defining the `assignable?` method, but I don't understand using `assignable?` when defining the `instance?` method.